### PR TITLE
Color Math

### PR DESCRIPTION
**Pull request by JoeJanickiChSh**
Adds a utility class for doing vector math with the Color object.

### CODE REVIEW

**Review by msoucy — CHANGES_REQUESTED** [review]
I'd also like to see unit tests added - especially for `lerp`

File: core/src/test/java/com/chopshop166/chopshoplib/ColorMathTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        final Color colorRight = new Color(0.3, 0.6, 0.9);
+        final double expected = 1.26;
+
+        assertTrue(Math.abs(expected - ColorMath.dot(colorLeft, colorRight)) <= EPSILON,
+                "Color dot product returns correct value");
+    }
+
+    /** Check color interpolation */
+    @Test
+    /* package */ void testLerp() {
```
Comment (msoucy) [inline core/src/test/java/com/chopshop166/chopshoplib/ColorMathTest.java:85]: Good unit test convention is to test edge cases as well - in other words, it would make sense to write a test for the midpoint (which you did), as well as interpolate with 0.0, 1.0, maybe even -0.5 and 1.5

**Review by msoucy — APPROVED** [review]

**Review by bot190 — APPROVED** [review]
